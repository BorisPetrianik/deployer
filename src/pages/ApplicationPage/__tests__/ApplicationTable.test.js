import React, { useState } from 'react'
import { makeStyles } from '@material-ui/core/styles'
// import { TableCell, TableRow, IconButton, Tooltip, CircularProgress, Grid } from '@material-ui/core'
// import { IApplicationInstance } from 'types/Application'
import PlayArrowIcon from '@material-ui/icons/PlayArrow'
import StopIcon from '@material-ui/icons/Stop'
import ReplayIcon from '@material-ui/icons/Replay'
import { shallow } from 'enzyme'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'
import { InstancesTable } from '../InstancesTable/InstancesTable'
import { TableCell, TableRow, IconButton, Tooltip, CircularProgress, Grid } from '@material-ui/core'
Enzyme.configure({ adapter: new Adapter() });



describe('InstancesTable', ()=>{
    let data = []
    beforeEach(()=> {
        data = []
    })
    it('renders correctly', () => {
        const CheckRow = renderer
          .create(<InstancesTable data={data}/>)
          .toJSON();
        expect(CheckRow).toMatchSnapshot();
      });

      it('Should contain at least 4 sells', () => {
        const wrapper = shallow(<InstancesTable data={data}/>);
        expect(wrapper.find(TableCell).length).toBeGreaterThan(3);
      });

      it('Should contain at least 4 sells', () => {
        const wrapper = shallow(<InstancesTable data={data}/>);
        expect(wrapper.find(TableCell).length).toBeLessThan(10);
      });

      // it('Should contain right hader', () => {
      //   const wrapper = shallow(<InstancesTable data={data}/>);
      //   expect(wrapper.find('h3').text()).toEqual('ApplicationPageTabdata');
      // });
})