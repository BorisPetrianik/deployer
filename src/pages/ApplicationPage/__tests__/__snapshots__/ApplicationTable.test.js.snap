// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstancesTable renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiTableContainer-root"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root makeStyles-table-1"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root makeStyles-row-3 MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            Version
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            Deploy Initiator
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            Status
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            Actions
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            Alias
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      />
    </table>
  </div>
</div>
`;
